charters/Architecture/charter.md

```diff
@@ -0,0 +1,35 @@
+# Architecture++
+
+## Description
+
+Architecture++ is responsible for the technical oversight of the Data Coordination Platform (DCP), including identifying and addressing *cross-cutting* architectural concerns across DCP components.
+
+The core team is composed of *Technical Leads* representing each DCP component to ensure a diversity of skills and views needed for community technical decision-making. 
+
+## In-scope
+Strategic responsibilities
+* Set overall **technical** direction for the DCP
+* Publish an annual Architecture RFC documenting technical debt that needs to be addressed during roadmap planning by DCP PM++
```

Comment:
Is **annual** a good cadence for the DCP Architecture RFC from the perspective of the technical leads? I'm curious because DCP PM still needs to explore/publish refinements to the planning process for 2019. 

And regarding **roadmap**, we still need to agree on a common term for the planning artifact that _identifies and expresses customer needs as a set of themes or goals_ similar in nature to [Rust](https://blog.rust-lang.org/2018/03/12/roadmap.html) and other open source projects that rally their communities. 

How about:

* Publish an Architecture RFC documenting technical debt that must be addressed in DCP PM product plans

Reply:
Regarding technical debt: I've been trying to identify the reason for some of my discomfort here, and I think it is down to the idea of universal DCP-wide technical debt. Individual charters will have differing requirements and therefore will be more or less comfortable with certain levels of technical debt in certain functionality, and should be free to make decisions as appropriate.

To give a specific example, ingest removed detailed auth plans from their phase 1 roadmap because it incurred considerable cost (due to complex ingest ownership requirements) without delivering proportionate value. Auth* was inserted as a DCP wide pilot requirement, which created scope creep for ingest without adding value to the ingest MVP (as ingest had already created a mitigation plan).

There's a delicate balance to strike between an idealised architecture RFC and a lowest common denominator RFC that doesn't create an imposition on any charter. From this document, I'm unclear on what the process is for determining what technical debt is in scope for the architecture group and what is in scope for individual charters / tech leads.

Reply:
@tburdett I will clarify this as `technical debt that impacts delivery of DCP as an integrated system and must be address during roadmap planning by PM...`.

Here's my reasoning:
1. Tech debt will occur between components, not just within them.
2. It's totally reasonable for teams to have their own thresholds for tech debt so long as individual tech debt doesn't keep DCP from delivering on its goals as an integrated system. If that debt does slow the whole project down, then it is, by nature, a group problem.